### PR TITLE
feat(workspaces, databases, collections, collection): allow to open shell with initial state; add shell entrypoints to tab headers

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch adds new entry points for shell to the tab headers when we are in multiple connections mode. We also now allow to open shell with some initial state and use this to apply the following logic:

- when opening shell from the collections list,  evaluate `use <db>` command on start
- when opening shell from the collection tab, evaluate `use <db>` command and set initial input to `db[<escaped coll name>].find()`

WIP, still waiting for browser-repl release and the previous PR to get merged

### CODE REVIEW

File: packages/compass-workspaces/src/types.ts
Comment (gribnoysup) [inline packages/compass-workspaces/src/types.ts:20]: My first idea is to allow to pass those props directy to the compass-shell, this gives us a lot of flexibility when opening shell, but means that this initial state needs to be formed in place when open method is called. Another approach we can take is to make those more specific and make shell the place that figures out what is the initial state that needs to be applied. Don't have a strong preference here which direction to take, so open to suggestions and happy to change this before merging.

File: packages/compass-preferences-model/src/react.ts
Comment (gribnoysup) [inline packages/compass-preferences-model/src/react.ts:62]: Seems like we're having a bunch of places where the single preference hook is used in groups of 2+, so I thought I'd add this one as an easier way to access multiple preferences at once

Comment (addaleax) [reply packages/compass-preferences-model/src/react.ts:62]: Yeah, so, just for context, the only reason this wasn't part of the original implementation was that this was pretty hard to type correctly – obviously not an issue if all the options in the group have the same type, though

Comment (gribnoysup) [reply packages/compass-preferences-model/src/react.ts:62]: Ah, interesting! I wonder, was this something that was resolved by typescript itself at some point? Seems like right now typescript is able to correctly derive the value types for each key when the keys are of mixed type

Comment (addaleax) [reply packages/compass-preferences-model/src/react.ts:62]: Yeah, this seems to be working perfectly fine 🙂

File: packages/compass-shell/src/components/compass-shell/tab-compass-shell.tsx
Comment (addaleax) [inline packages/compass-shell/src/components/compass-shell/tab-compass-shell.tsx:93]: TIL :)

**Review by addaleax — APPROVED** [review]
This is only waiting for a shell release, right?

**Comment by gribnoysup** [comment]
> This is only waiting for a shell release, right?

Yep! (Sorry, just noticed the comment) And now that stuff is landing I'm rebasing, but there are no other changes to the code. Thanks for the prompt review!

**Pull request merged**
